### PR TITLE
Fixed SiS Refresh Utility

### PR DESCRIPTION
**Pull request by ianmagenta**
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [ ] No
- [X] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Updates the `SessionTimeoutModal` component so that it no longer passes `null` to the `refresh` utility function. Also updates the `API_SIGN_IN_SERVICE_URL` function so that it can't generate strings with the word `null` at the end.

## Related issue(s)

[Jira Ticket VI-446](https://jira.devops.va.gov/browse/VI-446)

## Testing done

- A new unit test was added to the `SessionTimeoutModal` component and was run locally.
- Can be replicated by running `yarn test:unit src/platform/user/tests/authentication/components/SessionTimeoutModal.unit.spec.js`.

## What areas of the site does it impact?

This only impacts `SessionTimeoutModal.jsx` and `src/platform/utilities/oauth/constants.js`.

## Acceptance criteria

- [X] Verify the existence of the `v0/sign_in/refreshnull` bug.
- [X] Fix the code so that `refreshnull` is never allowed.

### CODE REVIEW

File: src/platform/user/authentication/components/SessionTimeoutModal.jsx [hunk 1]
```diff
@@ -28,8 +28,7 @@ export class SessionTimeoutModal extends React.Component {
   }
 
   componentDidUpdate() {
-    this.serviceName =
-      this.props.serviceName === undefined ? '' : this.props.serviceName;
+    this.serviceName = this.props.serviceName || '';
```
Comment (asg5704) [inline src/platform/user/authentication/components/SessionTimeoutModal.jsx:31]: 🤔  Can we change this?

Move the lookup inside the `mapStateToProps`

```javascript
// in componentDidUpdate
this.serviceName = this.props.serviceName;

// in mapStateToProps
// signInServiceNameSelector for SSOe, sessionStorage for SiS, fallback to empty string
const serviceName = signInServiceNameSelector(state) || sessionStorage.getItem('ci') || '';
return {
  // rest of code
  serviceName
 }
 ```

**Review by asg5704 — CHANGES_REQUESTED** [review]
Requested changes in the comment

File: src/platform/user/authentication/components/SessionTimeoutModal.jsx (hunk 1 above)
Comment (ianmagenta) [reply src/platform/user/authentication/components/SessionTimeoutModal.jsx:31]: Will do!

**Review by asg5704 — DISMISSED** [review]

**Comment by ianmagenta** [comment]
@asg5704 - Sorry, I need 1 more review. I had to push changes to the tests to reflect the updates in `SessionTimeoutModal`.

**Review by asg5704 — APPROVED** [review]

**Pull request merged**